### PR TITLE
Updates for THT reviews

### PR DESCRIPTION
**Pull request by elfiesmelfie**
Updates for the newest version of collectd-container-ansible.yaml  [1]

Main changes are:
* Regroup tasks
* Remove tasks/main and tasks/{{ deploy_stage }}
* Update tests to reflect this

### CODE REVIEW

**Comment by elfiesmelfie** [comment]
More details for the changes are in the commit messages, and explain the various changes.

File: .ansible-lint [hunk 1]
```diff
@@ -4,5 +4,6 @@ skip_list:
    - '303'  # systemctl used in place of systemd module
    - '701'  # Role info should contain platforms
 warn_list:
+   - '204'  # Lines should be no longer than 160 chars
```
Comment (csibbitt) [inline .ansible-lint:7]: They aren't going to go for this in tripleo-ansible, I spent all morning shortening my lines. You must pass their tox tests to be accepted and that includes yamllint and ansible-lint without our own exception rules.

https://github.com/openstack/tripleo-ansible/blob/master/.ansible-lint
https://github.com/openstack/tripleo-ansible/blob/master/tox.ini

File: filter_plugins/filter_snake_case.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# copied from https://github.com/infrawatch/tripleo-qdr-ansible-role/commit/4897cb6a09fd83c12b8d7a578c845471afe68dbc
+import re
+
+def filter_snake_case(camel_case_string):
+  name = re.sub(r'(.)([A-Z][a-z]+)', r'\1_\2', camel_case_string)
```
Comment (csibbitt) [inline filter_plugins/filter_snake_case.py]: There is actually a bug here, this should be `camel_case_string=...` See https://review.opendev.org/c/openstack/tripleo-ansible/+/772920/2/tripleo_ansible/roles/tripleo_metrics_qdr/filter_plugins/filter_snake_case.py

But also see https://review.opendev.org/c/openstack/tripleo-ansible/+/772920/1/tripleo_ansible/roles/tripleo_metrics_qdr/filter_plugins/filter_snake_case.py "Role filters should go in the (tripleo-ansible) filters directory

**Review by csibbitt — COMMENTED** [review]
Basically LGTM, but you might want to read Kevin's comments on my review to see what else could apply to you.

https://review.opendev.org/c/openstack/tripleo-ansible/+/772920/2#message-99670fb83178f13927d3026cb2b92e81169e4fe6

File: filter_plugins/filter_snake_case.py [hunk 3]
```diff
@@ -1,10 +1,14 @@
-# copied from https://github.com/infrawatch/tripleo-qdr-ansible-role/commit/4897cb6a09fd83c12b8d7a578c845471afe68dbc
+# copied from
```
Comment (csibbitt) [inline filter_plugins/filter_snake_case.py:1]: BTW as part of my review I've updated this again slightly and moved it into it's correct home (https://github.com/openstack/tripleo-ansible/tree/master/tripleo_ansible/ansible_plugins/filter). It hasn't been posted yet, but that will be the eventual fate of this filter.

File: meta/main.yml [hunk 4]
```diff
@@ -1,16 +1,33 @@
...
+  author: OpenStack
+  description: TripleO OpenStack role -- tripleo_collectd
+  company: Red Hat
   license: Apache-2.0
-
   min_ansible_version: 2.9
+  platforms:
+    - name: CentOS
+      versions:
+        - 7
```
Comment (csibbitt) [inline meta/main.yml:27]: Do we actually support 7?

**Review by csibbitt — APPROVED** [review]

File: filter_plugins/filter_snake_case.py (hunk 3 above)
Comment (elfiesmelfie) [reply filter_plugins/filter_snake_case.py:1]: Awesome, I'll make this file go away when I move to the tripleo_ansible repo!

File: meta/main.yml (hunk 4 above)
Comment (elfiesmelfie) [reply meta/main.yml:27]: That's anyone's guess, however, this is probably a copy-paste error

**Comment by elfiesmelfie** [comment]
The tests are failing because of podman, cos the molecule action doesn't support it. I'll remove the podman stuff from here, so I can continue to do some testing in an environment that I know, and when @csibbitt has figured out the process for updating the tests to run with test_infra, I'll follow his lead and update molecule then. Otherwise, I'm wasting time updating GHA for a very short time.

**Comment by csibbitt** [comment]
> The tests are failing because of podman, cos the molecule action doesn't support it. I'll remove the podman stuff from here, so I can continue to do some testing in an environment that I know, and when @csibbitt has figured out the process for updating the tests to run with test_infra, I'll follow his lead and update molecule then. Otherwise, I'm wasting time updating GHA for a very short time.

I didn't convert anything to testinfra since I have lots of coverage in ansible. I just changed the verifier to `ansible` and hopefully the folks at tripleo-ansible will be okay with it. :)

As for your podman trouble, you can work around it in github like this:
- https://github.com/infrawatch/tripleo-qdr-ansible-role/blob/master/molecule/default/molecule.yml#L3
- Then set `CONTAINER_BIN: docker` in your GHA workflow

**Pull request merged**